python/tests/pyspark/string_map_test.py

```diff
@@ -19,85 +22,113 @@
 
 class StringMapTest(unittest.TestCase):
 
-    def setUp(self):
-        self.spark = spark_session()
-        self.input = self.spark.createDataFrame([['a', 'b']], INPUT_SCHEMA)
+    @classmethod
+    def setUpClass(cls):
+        cls.spark = spark_session()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.spark.stop()
 
-    def tearDown(self):
-        self.spark.stop()
+    def setUp(self):
+        self.input = StringMapTest.spark.createDataFrame([['a', 'b']], INPUT_SCHEMA)
 
     def test_map(self):
-        result = StringMap({'a': 1.0}, 'key_col', 'value_col').transform(self.input)
-        expected = self.spark.createDataFrame([['a', 'b', 1.0]], OUTPUT_SCHEMA)
+        result = StringMap(
+            labels={'a': 1.0},
+            inputCol='key_col',
+            outputCol='value_col',
+        ).transform(self.input)
+        expected = StringMapTest.spark.createDataFrame([['a', 'b', 1.0]], OUTPUT_SCHEMA)
         assert_df(expected, result)
 
     def test_map_default_value(self):
-        result = StringMap({'z': 1.0}, 'key_col', 'value_col', handleInvalid='keep').transform(self.input)
-        expected = self.spark.createDataFrame([['a', 'b', 0.0]], OUTPUT_SCHEMA)
+        result = StringMap(
+            labels={'z': 1.0},
+            inputCol='key_col',
+            outputCol='value_col',
+            handleInvalid='keep',
+        ).transform(self.input)
+        expected = StringMapTest.spark.createDataFrame([['a', 'b', 0.0]], OUTPUT_SCHEMA)
         assert_df(expected, result)
 
     def test_map_custom_default_value(self):
-        result = StringMap({'z': 1.0}, 'key_col', 'value_col', handleInvalid='keep', defaultValue=-1.0) \
-            .transform(self.input)
-        expected = self.spark.createDataFrame([['a', 'b', -1.0]], OUTPUT_SCHEMA)
+        result = StringMap(
+            labels={'z': 1.0},
+            inputCol='key_col',
+            outputCol='value_col',
+            handleInvalid='keep',
+            defaultValue=-1.0
+        ).transform(self.input)
+        expected = StringMapTest.spark.createDataFrame([['a', 'b', -1.0]], OUTPUT_SCHEMA)
         assert_df(expected, result)
 
     def test_map_missing_value_error(self):
         with self.assertRaises(Py4JJavaError) as error:
-            StringMap({'z': 1.0}, 'key_col', 'value_col').transform(self.input).collect()
+            StringMap(
```

Comment:
nitpick, is the extra space required?

Reply:
What extra space? If you mean putting all args on separate lines, I'm following python linters.
It'd be nice to introduce linter checks in `make test` for the python subproject, I'll try get to it while doing the `MathBinary`